src/main/scala/net/fabricmc/language/scala/ScalaLanguageAdapter.scala

```diff
@@ -16,22 +16,42 @@
 
 package net.fabricmc.language.scala
 
-
+import net.fabricmc.language.scala.ScalaLanguageAdapter.logger
 import net.fabricmc.loader.api.{LanguageAdapter, ModContainer}
+import org.slf4j.LoggerFactory
+import scala.util.{Failure, Success, Try}
+import net.fabricmc.loader.api.LanguageAdapterException
 
 class ScalaLanguageAdapter extends LanguageAdapter {
 
-  override def create[T](modContainer: ModContainer, s: String, aClass: Class[T]): T = {
-    try {
-      val objectClass = Class.forName(s + "$")
-      val moduleField = objectClass.getField("MODULE$")
-      val instance = moduleField.get(null).asInstanceOf[T]
-      if (instance == null) throw new NullPointerException
-      instance
-    } catch {
-      case _: Exception =>
-        println(s"Unable to find ${aClass.getName}$$MODULE$$")
-        aClass.getConstructor().newInstance()
+    override def create[T](
+        modContainer: ModContainer,
+        value: String,
+        typ: Class[T]
+    ): T = {
+        // Load Object
+        val tried = Try {
+            val objectClass = Class.forName(value + "$")
+            val moduleField = objectClass.getField("MODULE$")
+            val instance = moduleField.get(null).asInstanceOf[T]
+            Option(instance).getOrElse(throw new NullPointerException)
+        }
+
+        tried match
+            case Success(instance) => return instance
+            case _                 =>
+
+        logger.warn(
+          s"Unable to find ${value}.MODULE$$. Maybe it isn't a Scala object?"
+        )
```

Comment:
I wonder if this should just be a fatal error?

Reply:
hum I think so, I'm trying to figure out a way to support normal scala class, not done with that though. In the future I will submit a new pr to support normal scala class